src/websocket/index.ts

```diff
@@ -191,21 +209,28 @@ export const onConnection = async (socket: WebSocket.WebSocket): Promise<void> =
 
   socket.on('close', (code, reason) => {
     // Clear timeout on close
-    clearTimeout(timeoutId)
+    clearTimeout(timeoutId);
 
     // Decrement connection counter
-    activeConnections--
+    activeConnections--;
 
-    console.log(`WebSocket connection closed with code: ${code} and reason: ${reason}`)
-    nestedCountersInstance.countEvent('websocket', 'close')
+    const currentIPConnections = connectionsByIP.get(ip);
+    if (currentIPConnections) {
+      currentIPConnections.delete(socket);
+      if (currentIPConnections.size === 0) {
+        connectionsByIP.delete(ip);
+      }
+    }
+    console.log(`WebSocket connection closed with code: ${code} and reason: ${reason}`);
+    nestedCountersInstance.countEvent('websocket', 'close');
     if (logSubscriptionList.getBySocket(socket)) {
       logSubscriptionList.getBySocket(socket)?.forEach((subscription_id) => {
         subscriptionEventEmitter.emit('evm_log_unsubscribe', subscription_id)
       })
       logSubscriptionList.removeBySocket(socket)
       socket.close(code, reason)
     }
-    if(CONFIG.verbose) console.log(logSubscriptionList.getAll())
+    if (CONFIG.verbose) console.log(logSubscriptionList.getAll());
```

Comment:
add more detail to this log message so we know whats being printed out.

Reply:
done